src/main/java/com/minecolonies/coremod/inventory/GuiRack.java

```diff
@@ -0,0 +1,66 @@
+package com.minecolonies.coremod.inventory;
+
+import com.minecolonies.coremod.tileentities.TileEntityRack;
+import net.minecraft.client.gui.inventory.GuiContainer;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+import net.minecraftforge.items.IItemHandler;
+
+@SideOnly(Side.CLIENT)
+public class GuiRack extends GuiContainer
+{
+    /**
+     * The resource location of the texture.
+     */
+    private static final ResourceLocation CHEST_GUI_TEXTURE = new ResourceLocation("textures/gui/container/generic_54.png");
+    /**
+     * The upper chest inventory.
+     */
+    private final IItemHandler upperChestInventory;
+
+    /**
+     * The lower chest inventory.
+     */
+    private final IItemHandler   lowerChestInventory;
+
+    /**
+     * Used to calculate the window height.
+     */
+    private int inventoryRows;
+
+    public GuiRack(final InventoryPlayer parInventoryPlayer, final TileEntityRack tileEntity, final TileEntityRack neighborRack, final World world, final BlockPos location)
+    {
+        super(new ContainerRack(tileEntity, neighborRack, parInventoryPlayer, world, location));
+        this.upperChestInventory = tileEntity.getInventory();
+        this.inventoryRows = upperChestInventory.getSlots() / 9;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 9 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)